plugins/experimental/buffer_upload/buffer_upload.cc

```diff
@@ -753,6 +753,15 @@ attach_pvc_plugin(TSCont /* contp ATS_UNUSED */, TSEvent event, void *edata)
         memcpy(replacement_host_str, host_hdr_str_val, host_hdr_str_val_len);
         TSDebug(DEBUG_TAG, "Adding host to request url: %s", replacement_host_str);
 
+        const char *colon = strstr(replacement_host_str, ":");
+        if (colon != NULL && colon + 1 != NULL) {
+          int port_str_val = atoi(colon + 1);
+
+          if (port_str_val != 80) {
+            TSUrlPortSet(req_bufp, url_loc, port_str_val);
+          }
+          host_hdr_str_val_len -= strlen(colon);
```

Comment:
Why not {{host_hdr_str_val_len = colon - host_hdr_str_val_len;}} if you want to clip at the colon?